### DISCUSSION TITLE
preact.config.js does not update babel config

### DISCUSSION THREAD

**Discussion by TbirdDuncan** [preactjs/preact-cli#1657] [Q&A]
Same as title, a  .babelrc.json does however work.

**Comment by rschristian** [comment]
How are you attempting to alter the config? We need a lot more information to help you. Vague titles are not problem descriptions.

You can alter the Babel config with the following:

`preact.config.js`
```js
export default (config, env, helpers) => {
    const { rule } = helpers.getLoadersByName(config, 'babel')[0];
    const babelConfig = rule.options;

    babelConfig.plugins.push(
        'my-babel-plugin'
    );
}
```

**Comment by TbirdDuncan** [comment]
Attempting to change the config to be conducive to jest testing as jest errors out when rendering due to jsx, @babel/preset-react is needed...

```
export default (config, env, helpers, options) => {
  /** you can change the config here **/
  let { rule } = helpers.getLoadersByName(config, "babel-loader")[0];
  let babelConfig = rule.options;

  babelConfig.presets.push("@babel/preset-react");
};

**Reply by rschristian** [reply]
Jest isn't ran with `preact-cli`. It's plain Jest. Jest doesn't care or pick up `preact.config.js`.

That being said, the `jest-preset-preact` handles that already.

**Reply by TbirdDuncan** [reply]
So I'm following this... [preact testing docs](https://preactjs.com/guide/v10/preact-testing-library). It doesn't work because when it goes to render that first component jest errors out and says it can render jsx. Where is this jest-preset-preact?

**Reply by rschristian** [reply]
It's added by default in all of the templates, or you can find the repo here: https://github.com/preactjs/jest-preset-preact

How you configure Jest is completely separate from Preact-CLI. Two different tools.

Edit: We certainly could do with more information there, but at the moment, it looks like we do mostly assume you can handle setting up Jest/alternatives yourself.

**Reply by TbirdDuncan** [reply]
hmm, but there was no jest script, I'm sorry this is not making much sense. I went and installed jest seperately, and it can run simple tests successfully but rendering components is no bueno currently. Should I just remove all associations with jest entirely and then... 🤔

**Reply by TbirdDuncan** [reply]
https://github.com/preactjs-templates/simple/blob/master/template/package.json

Here's the scripts from the simple template: 
```
  "scripts": {
    "build": "preact build",
    "serve": "sirv build --port 8080 --cors --single",
    "dev": "preact watch",
    "lint": "eslint src"
  },

**Reply by TbirdDuncan** [reply]
I figured jest had to be in there

**Reply by rschristian** [reply]
Simple has no testing setup. Part of the simplicity is that unnecessary items, like testing, have been cut out.

You can look at [the default template](https://github.com/preactjs-templates/default/tree/master/template) for a testing setup.

**Reply by TbirdDuncan** [reply]
Okay! I got her up and running thank you! 
Solution: 
>Deleted preact.config.js from root
Deleted .babelrc.json from root
updated package.json to: 
>> {
  "private": true,
  "name": "todo3",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "build": "preact build",
    "serve": "sirv build --port 8080 --cors --single",
    "dev": "preact watch",
    "lint": "eslint src",
    "test": "jest"
  },
  "eslintConfig": {
    "extends": "preact",
    "ignorePatterns": [
      "build/"
    ]
  },
  "devDependencies": {
    "@babel/core": "^7.17.0",
    "@babel/preset-env": "^7.16.11",
    "@babel/preset-react": "^7.16.7",
    "@testing-library/preact": "^2.0.1",
    "babel-jest": "^27.4.6",
    "eslint": "^7.17.0",
    "eslint-config-preact": "^1.3.0",
    "jest": "^27.4.7",
    "jest-preset-preact": "^4.0.5",
    "sirv-cli": "^1.0.3"
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.6.2",
    "axios": "^0.25.0",
    "core-js": "^3.21.0",
    "preact": "^10.1.0",
    "preact-cli": "^3.3.3",
    "preact-render-to-string": "^5.1.2",
    "react-redux": "^7.2.6",
    "redux-saga": "^1.1.3",
    "regenerator-runtime": "^0.13.9",
    "web-vitals": "^1.0.1"
  },
  "jest": {
    "preset": "jest-preset-preact"
  }
}


and finally in my tests I imported : 
>import "core-js/stable";
import "regenerator-runtime/runtime";
